### ISSUE TITLE
Crash: Parameter chat should be non-null

### ISSUE THREAD

**Issue by guillaume-tgl** [helpscout/beacon-android-sdk-sample#60]
* Beacon version: 1.0.5
* Android OS version: 8.0
* Device make/model: All

**Steps to reproduce**
1. Start Beacon with `BeaconActivity.open(context);`

**Actual outcome**
Crash with stacktrace:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.thegrizzlylabs.geniusscan.free.debug/com.helpscout.beacon.internal.domain.config.BeaconConfigActivity}: org.koin.core.error.InstanceCreationException: Could not create instance for [type:Factory,class:'com.helpscout.beacon.internal.common.l']
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3092)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3235)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1926)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:6986)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
     Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [type:Factory,class:'com.helpscout.beacon.internal.common.l']
        at org.koin.core.instance.DefinitionInstance.create(DefinitionInstance.kt:61)
        at org.koin.core.instance.FactoryDefinitionInstance.get(FactoryDefinitionInstance.kt:37)
        at org.koin.core.definition.BeanDefinition.resolveInstance(BeanDefinition.kt:75)
        at org.koin.core.Koin.createEagerInstances$koin_core(Koin.kt:125)
        at org.koin.core.KoinApplication.createEagerInstances(KoinApplication.kt:116)
        at org.koin.core.context.GlobalContextKt.startKoin(GlobalContext.kt:73)
        at com.helpscout.beacon.a.d.c$a.a(Unknown Source:21)
        at com.helpscout.beacon.internal.common.d.t(Unknown Source:11)
        at com.helpscout.beacon.internal.common.d.onCreate(Unknown Source:3)
        at com.helpscout.beacon.internal.domain.config.BeaconConfigActivity.onCreate(Unknown Source:0)
        at android.app.Activity.performCreate(Activity.java:7326)
        at android.app.Activity.performCreate(Activity.java:7317)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3072)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3235) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1926) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:6986) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445) 
     Caused by: org.koin.core.error.InstanceCreationException: Could not create instance for [type:Single,class:'com.helpscout.beacon.internal.common.a']
        at org.koin.core.instance.DefinitionInstance.create(DefinitionInstance.kt:61)
        at org.koin.core.instance.SingleDefinitionInstance.get(SingleDefinitionInstance.kt:40)
        at org.koin.core.definition.BeanDefinition.resolveInstance(BeanDefinition.kt:75)
        at org.koin.core.Koin.resolve(Koin.kt:108)
        at org.koin.core.Koin.get(Koin.kt:97)
        at com.helpscout.beacon.a.d.k.a(Unknown Source:25)
        at com.helpscout.beacon.a.d.k.invoke(Unknown Source:4)
        at org.koin.core.instance.DefinitionInstance.create(DefinitionInstance.kt:54)
        at org.koin.core.instance.FactoryDefinitionInstance.get(FactoryDefinitionInstance.kt:37) 
        at org.koin.core.definition.BeanDefinition.resolveInstance(BeanDefinition.kt:75) 
        at org.koin.core.Koin.createEagerInstances$koin_core(Koin.kt:125) 
        at org.koin.core.KoinApplication.createEagerInstances(KoinApplication.kt:116) 
        at org.koin.core.context.GlobalContextKt.startKoin(GlobalContext.kt:73) 
        at com.helpscout.beacon.a.d.c$a.a(Unknown Source:21) 
        at com.helpscout.beacon.internal.common.d.t(Unknown Source:11) 
        at com.helpscout.beacon.internal.common.d.onCreate(Unknown Source:3) 
        at com.helpscout.beacon.internal.domain.config.BeaconConfigActivity.onCreate(Unknown Source:0) 
        at android.app.Activity.performCreate(Activity.java:7326) 
        at android.app.Activity.performCreate(Activity.java:7317) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3072) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3235) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1926) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:6986) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445) 
     Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter chat
        at com.helpscout.beacon.internal.core.model.BeaconMessaging.<init>(Unknown Source:12)
        at com.helpscout.beacon.internal.core.model.BeaconConfig.applyLocalOverrides(ApiModels.kt:127)
        at com.helpscout.beacon.internal.core.model.BeaconConfig.withOverrides(ApiModels.kt:119)
        at com.helpscout.beacon.internal.core.data.BeaconPreferences.getConfigWithOverrides(BeaconPreferences.kt:270)
        at com.helpscout.beacon.internal.common.k.<init>(Unknown Source:15)
        at com.helpscout.beacon.a.d.h.a(Unknown Source:35)
        at com.helpscout.beacon.a.d.h.invoke(Unknown Source:4)
        at org.koin.core.instance.DefinitionInstance.create(DefinitionInstance.kt:54)
```

**Expected outcome**
No crash

**Comment by guillaume-tgl** [comment]
And here is how I configure my Beacon:
```
Beacon.setConfigOverrides(new BeaconConfigOverrides(true, true, false, contactForm, getBeaconColor()));
```

**Comment by jotaemepereira** [comment]
Hey @guillaume-tgl, thanks for detailed information and sorry for the 💥. We already created a ticket in our internal system, and as soon as we get more information, we will get back to you.

**Comment by guillaume-tgl** [comment]
Ok, let me know if you need more details.

**Comment by guillaume-tgl** [comment]
I've tried the Customisation Sample with our Beacon Id and it works fine. I've even removed all parameters and kept only the Beacon Builder and `BeaconActivity.open(context)` in both apps and it's still crashing in my app and not crashing in the demo app. I have no idea what else could have an impact.
Have you checked the SDK code to see how this could happen based on the stack trace I provided? I would gladly investigate but it's not open-source 😢 It seems like the crash happens when Koin tries to create an instance of a dependency.

**Comment by guillaume-tgl** [comment]
Calling `Beacon.clear()` does indeed prevent the crash. So you might be on the right track with this cache.

**Comment by scottyab** [comment]
Thanks for confirming. I've fixed it 👍  and it'll ship in v1.0.6. 

I think the issue is caused when`BeaconMessaging` object is being created as part of the Koin.init() by Moshi from a cached/stored JSON copy. Between version 1.0.4 and 1.0.5 we introduced a new a non-null kotlin param to `BeaconMessaging` called `chatConfig` and it seems the error occurs when restoring the config after upgrade to 1.0.5 as previously there was no value for `chatConfig`. Fresh installs of 1.0.5 didn't see this issue.  In an effort to ensure this doesn't happen again we're gonna add some additional upgrade testing to our pre-release regression testing.
